pkg/controllers/managementuser/healthsyncer/healthsyncer.go

```diff
@@ -85,6 +93,19 @@ func (h *HealthSyncer) getComponentStatus(cluster *v3.Cluster) error {
 	sort.Slice(cluster.Status.ComponentStatuses, func(i, j int) bool {
 		return cluster.Status.ComponentStatuses[i].Name < cluster.Status.ComponentStatuses[j].Name
 	})
+
+	if excludedComponents, ok := excludedComponentMap[cluster.Status.Driver]; ok {
+		for i := 0; i < len(excludedComponents); i++ {
+			for ii := 0; ii < len(cluster.Status.ComponentStatuses); ii++ {
+				if cluster.Status.ComponentStatuses[ii].Name == excludedComponents[i] {
+					cluster.Status.ComponentStatuses = append(cluster.Status.ComponentStatuses[:ii], cluster.Status.ComponentStatuses[ii+1:]...)
+					break
+				}
+			}
+
+		}
+	}
+
```

Comment:
Hey, just took a quick look and I think you can probably simplify this more by doing something along these lines (untested).

```
func init() {
	excludedComponentMap[providers.AKS] = map[string]struct{}{"controller-manager": struct{}{}, "scheduler": struct{}{}}
}

if excludedComponents, ok := excludedComponentMap[cluster.Status.Driver]; ok {
    for i, status := range cluster.Status.ComponentStatuses {
        if _, ok := excludedComponents[status.Name]; ok {
            cluster.Status.ComponentStatuses = append(cluster.StatusComponentStatuses[:i], cluster.Status.ComponentStatuses[i+1:]...)
            break
        }
}
```

Reply:
ill refactor it to use a map thanks!

Reply:
@PennyScissors I looked into refactoring to using a map and remembered why I did it this way.  If you delete an index from an array you are traversing you will get an Index error because the length will have decreased.  While we do traverse the list multiple times we reset our length in the event of a delete.  Most clusters only have 1 - 3 statues so this  should not be a performance hit.